pkg/controller/certificates/sync.go

```diff
@@ -361,6 +380,48 @@ func (c *Controller) issue(ctx context.Context, issuer issuer.Interface, crt *v1
 	return nil
 }
 
+var serialNumberLimit = new(big.Int).Lsh(big.NewInt(1), 128)
+
+func (c *Controller) generateSelfSignedExpiredCertificate(crt *v1alpha1.Certificate, pk []byte) ([]byte, error) {
+	signer, err := pki.DecodePrivateKeyBytes(pk)
+	if err != nil {
+		return nil, err
+	}
+
+	pubKeyAlgo, _, err := pki.SignatureAlgorithm(crt)
+	if err != nil {
+		return nil, err
+	}
+
+	serialNumber, err := rand.Int(rand.Reader, serialNumberLimit)
+	if err != nil {
+		return nil, fmt.Errorf("failed to generate serial number: %s", err.Error())
+	}
+
+	now := c.clock.Now()
+	template := &x509.Certificate{
+		Version:               3,
+		BasicConstraintsValid: true,
+		SerialNumber:          serialNumber,
+		PublicKeyAlgorithm:    pubKeyAlgo,
+		Subject: pkix.Name{
+			Organization: []string{"cert-manager"},
+			CommonName:   "cert-manager.local",
```

Comment:
slightly worried that ingress controllers will do some SNI trickery and refuse to present this cert